src/AssetPicker/AssetPicker.tsx

```diff
@@ -0,0 +1,100 @@
+import '@clarabridge/unified-ui/dist/utilities.css';
+import React, { PropsWithChildren, ReactElement, useEffect, useState } from 'react';
+import { createPortal } from 'react-dom';
+import AssetPickerContent from './AssetPickerContent';
+import { Asset } from './Asset';
+import AssetPickerInput from './AssetPickerInput';
+import useFetch, { CachePolicies, IncomingOptions } from 'use-http';
+
+export type AssetPickerProps = PropsWithChildren<{
+    isOpen: boolean;
+    container?: Element;
+    onSelected?: (asset: Asset) => void;
+    extraText?: string;
+}>;
+
+export default function AssetPicker(props: AssetPickerProps): ReactElement | null {
+    const limit = 10;
+
+    const [offset, setOffset] = useState(0);
+    const [keyword, setKeyword] = useState('');
+    const [oldKeyword, setOldKeyword] = useState('');
+    
+    const options: IncomingOptions = {
+        headers: {
+            'Authorization' : 'Basic ' + btoa('Aimanoshi:Azerty123'),
+            'Content-Type' : 'application/json'
+        },
+        cachePolicy: CachePolicies.NO_CACHE,
+        perPage: limit
+    };
+
+    const url = 'http://localhost:1234/api/giphy';
+
+    const searchParams = new URLSearchParams();
+
+    if (offset) searchParams.append('offset', offset.toString());
+    if (limit) searchParams.append('limit', limit.toString());
+
+    let segment = '';
+    if (searchParams.toString().length) {
+        segment = `?${searchParams.toString()}`
+    }  
+    const [assets, setAssets] = useState<Asset[]>([]);
+    const { get, response , loading, error } = useFetch(url, options);
+    useEffect(() => { loadInitialAssets() }, [keyword,offset])
+
+    async function loadInitialAssets() {
+        const trendingGifs = await get(( keyword !== '' ? `/search/${keyword}` : '/trending') + segment);
+        if (oldKeyword !== keyword){
+            setOldKeyword(keyword);
+            setAssets([]);
+        }
+        if (trendingGifs.assets === undefined) { return setAssets(oldArray => [...oldArray])}
```

Comment:
Multilining this is a bit more clear

```suggestion
        if (trendingGifs.assets === undefined) { 
            return setAssets(oldArray => [...oldArray]);
        }
```